lib/CXGN/Chado/Stock.pm

```diff
@@ -630,20 +630,87 @@ sub get_trials {
     my $geolocation_type_id = SGN::Model::Cvterm->get_cvterm_row($self->get_schema(), 'project location', 'project_property')->cvterm_id();
     my $q;
     if ($stock_type eq 'accession'){
-        $q = "select distinct(project.project_id), project.name, projectprop.value from stock as accession join stock_relationship on (accession.stock_id=stock_relationship.object_id) JOIN stock as plot on (plot.stock_id=stock_relationship.subject_id) JOIN nd_experiment_stock ON (plot.stock_id=nd_experiment_stock.stock_id) JOIN nd_experiment_project USING(nd_experiment_id) JOIN project USING (project_id) FULL OUTER JOIN projectprop ON (project.project_id=projectprop.project_id AND projectprop.type_id=$geolocation_type_id) WHERE accession.stock_id=?;";
-    } else {
-        $q = "select distinct(project.project_id), project.name, projectprop.value from stock JOIN nd_experiment_stock USING(stock_id) JOIN nd_experiment_project USING(nd_experiment_id) JOIN project USING (project_id) FULL OUTER JOIN projectprop ON (project.project_id=projectprop.project_id AND projectprop.type_id=$geolocation_type_id) WHERE stock.stock_id=?;";
+        $q = "SELECT DISTINCT project.project_id, project.name, projectprop.value 
+        FROM stock AS accession 
+        JOIN stock_relationship ON (accession.stock_id=stock_relationship.object_id) 
+        JOIN stock AS plot ON (plot.stock_id=stock_relationship.subject_id) 
+        JOIN nd_experiment_stock ON (plot.stock_id=nd_experiment_stock.stock_id) 
+        JOIN nd_experiment_project USING (nd_experiment_id) 
+        JOIN project USING (project_id) 
+        FULL OUTER JOIN projectprop ON (project.project_id=projectprop.project_id AND projectprop.type_id=$geolocation_type_id) 
+        JOIN cvterm ON (cvterm.cvterm_id=nd_experiment_stock.type_id) 
+        FULL OUTER JOIN breeding_programs ON (breeding_programs.breeding_program_id=project.project_id) 
+        WHERE accession.stock_id=? 
+            AND cvterm.name!='phenotyping_experiment' 
+            AND cvterm.name!='analysis_experiment' 
+            AND breeding_program_id IS NULL;";
+    } else { # Odds are if this is not an accession, it is a plot.
```

Comment:
can it also be a plant or tissue sample? is the query going to work in such cases?

Reply:
I am pretty sure it would not work if those stocks were used but the original code that was there only distinguishes between accessions and non-accession stocks, and I did not want to change that. As far as I can tell though, these functions are only ever called on accessions and plots, so I do not think it matters. Would it be better to throw an error for invalid stock types?

Reply:
I think it would be better to be explicit about this, and check if the stock is an accession or a plot, and throw an error if neither